extensions/amp-video/0.1/amp-video.js

```diff
@@ -192,6 +192,11 @@ class AmpVideo extends AMP.BaseElement {
     const {element} = this;
 
     this.configure_();
+    this.intersectionObserver_ = new IntersectionObserver(entries => {
```

Comment:
IMHO, we need to find a way to safely share InOb instance for elements in the same window or same ampdoc. But carefully. We can only share an instance of InOb for the same purpose. Otherwise, we'd have race conditions related to someone unobserving an element before other users still need it. I think in practice this could be simply resolved by doing a `devAssert` on whether an element has already been subscribed, to avoid overlapping subscribers.

Reply:
its an issue if you call `.observe` twice on the same elem? Im surprised thats not a noop. I agree about sharing an InOb if they have the same cb. The superseding PR does that

Reply:
It is a noop. The problem is with `unobserve`. E.g. imagine two overlapping use cases that use the same InOb for the same element:

```
1: |observe(el)-------------------------unobserve(el)|
2: |------|observe(el)------unobserve(el)|--xxxxxxx--|
```

The timeline marked with "x" is the problem. Essentially the second observer breaks the first one.